cli/runtime/config/config_factory.go

```diff
@@ -26,6 +26,11 @@ var LegacyConfigNodeKeys = []string{
 	KeyCurrentServer,
 }
 
+var DiscardedConfigNodeKeys = []string{
```

Comment:
nit: it's good to have some trace back to reason for the change. 
A comment explaining why might be good. 

Also: please add
```
Fixes #4072 
```
into the commit you plan to retain, that way there is linkage in the git commit history as well.

Fixes: #